isisd/isis_sr.c

```diff
@@ -0,0 +1,1654 @@
+	if (srp->local_label != MPLS_INVALID_LABEL)
+		label2str(srp->local_label, buf_llabel, sizeof(buf_llabel));
+	else
+		snprintf(buf_llabel, sizeof(buf_llabel), "-");
+
+	if (!srp->u.remote.rinfo) {
+		ttable_add_row(tt, "%s|%u|%s|-|-|-", buf_prefix, srp->sid.value,
+			       buf_llabel);
+		return;
+	}
+
+	for (ALL_LIST_ELEMENTS_RO(srp->u.remote.rinfo->nexthops, node,
+				  nexthop)) {
+		struct interface *ifp;
+
+		inet_ntop(nexthop->family, &nexthop->ip, buf_nhop,
+			  sizeof(buf_nhop));
+		ifp = if_lookup_by_index(nexthop->ifindex, VRF_DEFAULT);
+		if (ifp)
+			strlcpy(buf_iface, ifp->name, sizeof(buf_iface));
+		else
+			snprintf(buf_iface, sizeof(buf_iface), "ifindex %u",
+				 nexthop->ifindex);
+		if (nexthop->sr.label == MPLS_INVALID_LABEL) {
+			snprintf(buf_rlabel, sizeof(buf_rlabel), "-");
+			snprintf(buf_uptime, sizeof(buf_uptime), "-");
+		} else {
+			label2str(nexthop->sr.label, buf_rlabel,
+				  sizeof(buf_rlabel));
+			log_uptime(nexthop->sr.uptime, buf_uptime,
+				   sizeof(buf_uptime));
+		}
+
+		if (first)
+			ttable_add_row(tt, "%s|%u|%s|%s, %s|%s|%s", buf_prefix,
+				       srp->sid.value, buf_llabel, buf_nhop,
+				       buf_iface, buf_rlabel, buf_uptime);
+		else
+			ttable_add_row(tt, "|||%s, %s|%s|%s", buf_nhop,
+				       buf_iface, buf_rlabel, buf_uptime);
+		first = false;
+	}
+}
+
+static void isis_sr_show_prefix_sids(struct vty *vty, struct isis_area *area,
+				     int level)
+{
+	struct sr_prefix *srp;
+	struct ttable *tt;
+
+	if (tree_sr_area_prefix_count(&area->srdb.prefix_sids[level - 1]) == 0)
+		return;
+
+	vty_out(vty, " IS-IS %s Prefix-SIDs:\n\n", circuit_t2string(level));
+
+	/* Prepare table. */
+	tt = ttable_new(&ttable_styles[TTSTYLE_BLANK]);
+	ttable_add_row(tt, "Prefix|SID|In Label|Nexthop|Out Label|Uptime");
+	tt->style.cell.rpad = 2;
+	tt->style.corner = '+';
+	ttable_restyle(tt);
+	ttable_rowseps(tt, 0, BOTTOM, true, '-');
+
+	frr_each (tree_sr_area_prefix, &area->srdb.prefix_sids[level - 1],
+		  srp) {
+		switch (srp->type) {
+		case ISIS_SR_PREFIX_LOCAL:
+			isis_sr_show_prefix_sid_local(vty, tt, area, srp);
+			break;
+		case ISIS_SR_PREFIX_REMOTE:
+			isis_sr_show_prefix_sid_remote(vty, tt, area, srp);
+			break;
+		}
+	}
+
+	/* Dump the generated table. */
+	if (tt->nrows > 1) {
+		char *table;
+
+		table = ttable_dump(tt, "\n");
+		vty_out(vty, "%s\n", table);
+		XFREE(MTYPE_TMP, table);
+	}
+	ttable_del(tt);
+}
+
+DEFUN(show_sr_prefix_sids, show_sr_prefix_sids_cmd,
+      "show isis segment-routing prefix-sids",
+      SHOW_STR PROTO_HELP
+      "Segment-Routing\n"
+      "Segment-Routing Prefix-SIDs\n")
+{
+	struct listnode *node;
+	struct isis_area *area;
+
+	for (ALL_LIST_ELEMENTS_RO(isis->area_list, node, area)) {
+		vty_out(vty, "Area %s:\n",
+			area->area_tag ? area->area_tag : "null");
+
+		for (int level = ISIS_LEVEL1; level <= ISIS_LEVELS; level++)
+			isis_sr_show_prefix_sids(vty, area, level);
+	}
+
+	return CMD_SUCCESS;
+}
+
```

Comment:
New show commands print only prefix information disregarding the advertised SR-Node. SR-Node information, in particular SRGB is essential for debugging especially to detect SRGB overlap, index out of SRGB ...

Reply:
Since the table already shows the SID and input/output labels of each nexthop, I don't think the full SRGB information is necessary there. I'd prefer to not add any new column to the table to prevent it from getting too wide. The operators will always have `show isis database detail` as a last resort if they need full detailed information.